src/System.Private.TypeLoader/src/Internal/Reflection/Execution/AssemblyBinderImplementation.Ecma.cs

```diff
@@ -0,0 +1,194 @@
+        {
+            return reader.GetString(handle);
+        }
+
+        public static string GetStringOrNull(this StringHandle handle, MetadataReader reader)
+        {
+            if (handle.IsNil)
+                return null;
+
+            return reader.GetString(handle);
+        }
+
+        public static RuntimeAssemblyName ToRuntimeAssemblyName(this AssemblyDefinition assemblyDefinition, MetadataReader reader)
+        {
+            return CreateRuntimeAssemblyNameFromMetadata(
+                reader,
+                assemblyDefinition.Name,
+                assemblyDefinition.Version,
+                assemblyDefinition.Culture,
+                assemblyDefinition.PublicKey,
+                assemblyDefinition.Flags
+                );
+        }
+
+        public static RuntimeAssemblyName ToRuntimeAssemblyName(this AssemblyReferenceHandle assemblyReferenceHandle, MetadataReader reader)
+        {
+            AssemblyReference assemblyReference = reader.GetAssemblyReference(assemblyReferenceHandle);
+            return CreateRuntimeAssemblyNameFromMetadata(
+                reader,
+                assemblyReference.Name,
+                assemblyReference.Version,
+                assemblyReference.Culture,
+                assemblyReference.PublicKeyOrToken,
+                assemblyReference.Flags
+                );
+        }
+
+        private static RuntimeAssemblyName CreateRuntimeAssemblyNameFromMetadata(
+            MetadataReader reader,
+            StringHandle name,
+            Version version,
+            StringHandle culture,
+            BlobHandle publicKeyOrToken,
+            AssemblyFlags assemblyFlags)
+        {
+            AssemblyNameFlags assemblyNameFlags = AssemblyNameFlags.None;
+            if (0 != (assemblyFlags & AssemblyFlags.PublicKey))
+                assemblyNameFlags |= AssemblyNameFlags.PublicKey;
+            if (0 != (assemblyFlags & AssemblyFlags.Retargetable))
+                assemblyNameFlags |= AssemblyNameFlags.Retargetable;
+            int contentType = ((int)assemblyFlags) & 0x00000E00;
+            assemblyNameFlags |= (AssemblyNameFlags)contentType;
+
+            byte[] publicKeyOrTokenByteArray = null;
+            if (!publicKeyOrToken.IsNil)
+            {
+                ImmutableArray<byte> publicKeyOrTokenBlob = reader.GetBlobContent(publicKeyOrToken);
+                publicKeyOrTokenByteArray = new byte[publicKeyOrTokenBlob.Length];
+                publicKeyOrTokenBlob.CopyTo(publicKeyOrTokenByteArray);
+            }
+            
+            return new RuntimeAssemblyName(
+                name.GetString(reader),
+                version,
+                culture.GetString(reader),
+                assemblyNameFlags,
+                publicKeyOrTokenByteArray
+                );
+        }
+    }
+}
+
+namespace Internal.Reflection.Execution
+{
+    /// Abstraction to hold PE data for an ECMA module
+    public class PEInfo
```

Comment:
In the code as written... yes, but it isn't that useful in the places its used. Now that there was feedback to make an EcmaModuleInfo which I've taken, I shall add direct fields on that type as the size of the ModuleInfo is no longer an even minor concern, and the existing practice doesn't really provide value.